### PR TITLE
Add dbg mode for remote testing

### PR DESCRIPTION
**Pull request by adelez**

### CODE REVIEW

File: tools/internal_ci/linux/grpc_bazel_on_foundry_opt.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
```
Comment (matt-kwong) [inline tools/internal_ci/linux/grpc_bazel_on_foundry_opt.sh]: I don't think this file has executable permissions.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

File: tools/internal_ci/linux/grpc_bazel_on_foundry_opt.sh (hunk 1 above)
Comment (adelez) [reply tools/internal_ci/linux/grpc_bazel_on_foundry_opt.sh]: Copied from tools/jenkins/run_bazel_full.sh.

Comment (matt-kwong) [reply tools/internal_ci/linux/grpc_bazel_on_foundry_opt.sh]: I meant like running `chmod +x tools/internal_ci/linux/grpc_bazel_on_foundry_opt.sh`. GitHub is showing this file as not executable.

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

File: tools/internal_ci/linux/grpc_bazel_on_foundry_opt.sh (hunk 1 above)
Comment (adelez) [reply tools/internal_ci/linux/grpc_bazel_on_foundry_opt.sh]: Discussed offline. It works as is for the experiments. Will fix if something breaks.

**Comment by adelez** [comment]
Basic Tests MacOS [dbg]: #13669, #13681
Basic Tests MacOS [opt]: #13669
Basic Tests Multi-language Linux: #13669
Tsan C++: #13680

**Pull request merged**